### ISSUE TITLE
Remove npm from package.json dependencies

### ISSUE THREAD

**Issue by 3nuc** [0xfe/vexflow#720]
When you install Vexflow (`npm install vexflow`), you will get this message:

```bash
found 21 high severity vulnerabilities
run `npm audit fix` to fix them, or `npm audit` for details 
```

This is exactly the same issue as #653 . Just related to a different version of npm.

Let's run `npm audit` to see what is causing this message about vulnerabilities:
![image](https://user-images.githubusercontent.com/5359825/68082777-00a99380-fe21-11e9-9da5-d4fb3853299c.png)

As you can see in the `Path` row, `vexflow > npm`, this is caused by an old npm version that relies on a version of `https-proxy-agent` package whose version is older than 2.2.3.

This is caused by this line in package.json:

```
  "dependencies": {
    "npm": "^6.5.0"
  }
```

Solutions:
1. Update npm in package.json like #653 did
2. Move npm to devDependencies
3.  ~~(**BEST**) Remove npm from package.json completely and never have to worry about something like this again. Anybody that has downloaded the vexflow project NEEDS to have npm installed anyway, because they wouldn't be able to download the devDependencies for vexflow.~~

npm is downloaded based on package.json, but everybody has npm installed because they installed Node.js (npm comes with Node.js), so why keep it in package.json?

**Comment by 3nuc** [comment]
Actually, it seems like the situation is not so simple. I just cloned the vexflow repo and on newest Node LTS (12.13.0) it's impossible to run `npm install` in the vexflow repo without errors.

I downgraded to Node 8 & npm 6.4.1 and it installs correctly.

Still, if the prerequisite of running `npm i` in the vexflow repo is having Node 8 and npm 6.4.1 then there's no point in keeping the npm package in package.json.
